ipu-plugin/pkg/p4rtclient/rh_p4client.go

```diff
@@ -136,3 +137,202 @@ func (p *rhP4Client) getDelRuleSets(macAddr []byte, vlan int) []fxpRuleParams {
+* to configure the FXP with point to point rules between the VFs initialised on a single host.
+*
+* These rules assume that no NF has been deployed on the FXP.
+*
+* Function CreateAddPointToPointVFRules will create all the point to point rules between all the initilised VFs on the host.
+* Function DelFxpRules will remove all the point to point rules between all the initilised VFs on the host.
+ */
+func CreateAddPointToPointVFRules(p4rtbin string, vfMacList []string) {
+
+	ruleSets := []fxpRuleParams{}
+
+	for i := range vfMacList {
+		for j := range vfMacList {
+			if i != j {
+
+				srcVfMac, err := utils.GetMacAsByteArray(vfMacList[i])
+				if err != nil {
+					fmt.Printf("unable to extract octets from %s: %v", vfMacList[i], err)
+					return
+				}
+
+				dstVfMac, err := utils.GetMacAsByteArray(vfMacList[j])
+				if err != nil {
+					fmt.Printf("unable to extract octets from %s: %v", vfMacList[j], err)
+					return
+				}
+
+				dmac := strings.Replace(vfMacList[j], string(':'), "", -1)
+
+				ruleSets = append(ruleSets,
+					[]string{"add-entry", "br0", "rh_mvp_control.ingress_loopback_table",
+						fmt.Sprintf("vsi=0x%X,target_vsi=0x%X,action=rh_mvp_control.fwd_to_port(%d)", srcVfMac[1], dstVfMac[1], dstVfMac[1]+16)},
+					[]string{"add-entry", "br0", "rh_mvp_control.vport_egress_dmac_vsi_table",
+						fmt.Sprintf("vsi=0x%X,dmac=0x%s,action=rh_mvp_control.fwd_to_port(%d)", srcVfMac[1], dmac, dstVfMac[1]+16)},
+				)
+			}
+		}
+	}
+
+	for _, r := range ruleSets {
+		if err := utils.RunP4rtCtlCommand(p4rtbin, r...); err != nil {
+			log.WithField("error", err).Errorf("error executing del rule command")
+		} else {
+			log.Printf("Finished running: %s", p4rtbin+" "+strings.Join(r, " "))
+		}
+	}
+}
+
+func CreateDelPointToPointVFRules(p4rtbin string, vfMacList []string) {
+
+	ruleSets := []fxpRuleParams{}
+
+	for i := range vfMacList {
+		for j := range vfMacList {
+			if i != j {
+
+				srcVfMac, err := utils.GetMacAsByteArray(vfMacList[i])
+				if err != nil {
+					fmt.Printf("unable to extract octets from %s: %v", vfMacList[i], err)
+					return
+				}
+
+				dstVfMac, err := utils.GetMacAsByteArray(vfMacList[j])
+				if err != nil {
+					fmt.Printf("unable to extract octets from %s: %v", vfMacList[j], err)
+					return
+				}
+
+				dmac := strings.Replace(vfMacList[j], string(':'), "", -1)
+
+				ruleSets = append(ruleSets,
+					[]string{"del-entry", "br0", "rh_mvp_control.ingress_loopback_table",
+						fmt.Sprintf("vsi=0x%X,target_vsi=0x%X", srcVfMac[1], dstVfMac[1])},
+					[]string{"del-entry", "br0", "rh_mvp_control.vport_egress_dmac_vsi_table",
+						fmt.Sprintf("vsi=0x%X,dmac=0x%s", srcVfMac[1], dmac)},
+				)
+			}
+		}
+	}
+
+	for _, r := range ruleSets {
+		if err := utils.RunP4rtCtlCommand(p4rtbin, r...); err != nil {
+			log.WithField("error", err).Errorf("error executing del rule command")
+		} else {
+			log.Printf("Finished running: %s", p4rtbin+" "+strings.Join(r, " "))
```

Comment:
We should change this to either Printf with either `Infof()` or `Debugf()`.

Reply:
Addressed. Now using Infof() to show log in the console.